addons/mail/static/src/core/common/composer.xml

```diff
@@ -53,7 +53,7 @@
                             t-on-paste="onPaste"
                             t-model="props.composer.text"
                             t-att-placeholder="placeholder"
-                            t-att-readOnly="!state.active"
+                            t-att-readOnly="!state.active || thread?.read_only and !store.self.isAdmin"
```

Comment:
This check is repeated in several places, it would be nice to have a single place the centralize the logic.